amida-tech/hl7-telescoper-api-server
# added module to parse hl7 message and moved test data files into new …

### Related JIRA tickets:
>Please enter the whole URL so we can just click/copy it
- https://jira.amida.com/browse/HL7-12

### What exactly does this PR do?
>i.e. Added logic to do the thing because the thing didn't exist before. Please provide as much detail as possible. 
- Implements parsing of HL7 messages using the [health-level-seven-parser
](https://www.npmjs.com/package/health-level-seven-parser)

### What else was added outside of the scope of the ask?
- Moved some test files into a new test data directory. `server/tests/data`

### Do ANY of these changes introduce a breaking change? (in-scope or otherwise)
>i.e. Users need to update their `.env` files with the following... etc
- Will need to install health-level-seven-parser: `yarn add health-level-seven-parser`

### Manual test cases?
>Steps to manually test introduced changes can go here.
>Remember the prerequisites!
>Remember edge-cases you've caught in your code, etc..
> i.e. ignoring the button and clicking on the "NEXT" button should trigger a warning event because the thing that clicking the new button does didn't happen
- Start up your API server
- Post a file with HL7 message(s) to `http://localhost:4040/api/hl7/upload` (remember you'll need to signup and login etc). 
- Once its posted, look in your local mongo DB to make sure the `messages` collection has the parsed message field populated. 

### Any additional context/background?
>N/A if this is straight-forward
- N/A

### Screenshots (if appropriate):
